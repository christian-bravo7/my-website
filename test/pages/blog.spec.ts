import { createLocalVue, mount } from '@vue/test-utils';
import Blog from '@/pages/blog.vue';
import PortalVue from 'portal-vue';
import VueMeta from 'vue-meta';
import MockDate from 'mockdate';

const localVue = createLocalVue();
localVue.use(PortalVue);
localVue.use(VueMeta, { keyName: 'head' });

describe('blog page', () => {
  it('should render correctly', () => {
    const wrapper = mount(Blog, {
      localVue,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should present a greeting word properly for morning', () => {
    MockDate.set('Wed Nov 25 2020 1:00:00 GMT-0600 (Central Standard Time)');

    const wrapper = mount(Blog, {
      localVue,
    });

    // @ts-ignore
    const greeting = wrapper.vm.greetingWord;

    expect(greeting).toBe('good-morning');
  });

  it('should present a greeting word properly before noon', () => {
    MockDate.set('Wed Nov 25 2020 10:00:00 GMT-0600 (Central Standard Time)');

    const wrapper = mount(Blog, {
      localVue,
    });

    // @ts-ignore
    const greeting = wrapper.vm.greetingWord;

    expect(greeting).toBe('good-morning');
  });

  it('should present a greeting word properly for noon', () => {
    MockDate.set('Wed Nov 25 2020 12:00:00 GMT-0600 (Central Standard Time)');

    const wrapper = mount(Blog, {
      localVue,
    });

    // @ts-ignore
    const greeting = wrapper.vm.greetingWord;

    expect(greeting).toBe('good-noon');
  });

  it('should present a greeting word properly for afternoon', () => {
    MockDate.set('Wed Nov 25 2020 16:00:00 GMT-0600 (Central Standard Time)');

    const wrapper = mount(Blog, {
      localVue,
    });

    // @ts-ignore
    const greeting = wrapper.vm.greetingWord;

    expect(greeting).toBe('good-afternoon');
  });

  it('should present a greeting word properly for evening', () => {
    MockDate.set('Wed Nov 25 2020 20:00:00 GMT-0600 (Central Standard Time)');

    const wrapper = mount(Blog, {
      localVue,
    });

    // @ts-ignore
    const greeting = wrapper.vm.greetingWord;

    expect(greeting).toBe('good-evening');
  });

  it('should present a greeting word properly for night', () => {
    MockDate.set('Wed Nov 25 2020 23:00:00 GMT-0600 (Central Standard Time)');

    const wrapper = mount(Blog, {
      localVue,
    });

    // @ts-ignore
    const greeting = wrapper.vm.greetingWord;

    expect(greeting).toBe('good-evening');
  });

  it('should return dynamic class for post index', () => {
    const wrapper = mount(Blog, {
      localVue,
    });

    // @ts-ignore
    const dynamicClass = wrapper.vm.dynamicColors(1);

    expect(dynamicClass).toBe('hover:bg-pink-100 dark-hover:bg-pink-900');
  });

  it('should display meta-tag title', () => {
    const wrapper = mount(Blog, {
      localVue,
    });

    const title = wrapper.vm.$meta().refresh().metaInfo.title;

    expect(title).toBe('navigation.blog-label');
  });
});
