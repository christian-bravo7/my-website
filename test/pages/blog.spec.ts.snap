// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog page should render correctly 1`] = `
<section>
  <div class="container mx-auto py-20">
    <div class="flex flex-col items-center | mb-12">
      <h1 class="text-7xl font-bold | mb-2">
        Good morning
      </h1>
      <p class="text-lg">
        Welcome to my space
      </p>
    </div>
    <div class="flex justify-center"><label for="post-search" class="relative | w-full max-w-xl"><span class="absolute | left-4 top-1/2 | text-gray-400 | transform -translate-y-1/2 | select-none cursor-text | material-icons">
    search
  </span> <input id="post-search" type="text" placeholder="Search..." class="w-full | py-4 px-14 | bg-white | rounded-md | shadow-md"> <button class="absolute | right-4 top-1/2 | text-gray-400 | transform -translate-y-1/2 | select-none cursor-pointer | material-icons hidden">
          cancel
        </button></label></div>
  </div>
  <div class="container mx-auto px-4 md:px-0">
    <div class="blog-grid"></div>
  </div>
</section>
`;
